internal/manifests/configmap/configmap.go

```diff
@@ -0,0 +1,135 @@
+package configmap
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/ViaQ/logerr/kverrors"
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/elasticsearch-operator/internal/manifests/status"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CompareFunc is the type for functions that compare two configmaps.
+// Return true if two configmaps are equal.
+type CompareFunc func(current, desired *corev1.ConfigMap) bool
+
+// MutateFunc is the type for functions that mutate the current configmap
+// by applying the values from the desired configmap.
+type MutateFunc func(current, desired *corev1.ConfigMap)
+
+// Get returns the k8s configmap for the given object key or an error.
+func Get(ctx context.Context, c client.Client, key client.ObjectKey) (*corev1.ConfigMap, error) {
```

Comment:
no need to take in a context

Reply:
Of course it needs to get context here, as it is request by the k8s get call.

Reply:
see above comment